### ISSUE TITLE
java.lang.NoSuchMethodError: android.app.PendingIntent.getCreatorPackage

### ISSUE THREAD

**Issue by skypanther** [morinel/gcmpush#40]
User with Android 4.1.2 is getting a crash with the above error message. As shown in Appcelerator (crittercism) crash logs:

```
0   java.lang.NoSuchMethodError: android.app.PendingIntent.getCreatorPackage
1   at nl.vanvianen.android.gcm.GCMIntentService.onMessage(GCMIntentService.java:453)
2   at com.google.android.gcm.GCMBaseIntentService.onHandleIntent(GCMBaseIntentService.java:223)
3   at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:65)
4   at android.os.Handler.dispatchMessage(Handler.java:99)
5   at android.os.Looper.loop(Looper.java:137)
6   at android.os.HandlerThread.run(HandlerThread.java:60)
```

I've scanned the code and don't see this method being directly called, nor is [getTargetPackage()](https://developer.android.com/reference/android/app/PendingIntent.html#getTargetPackage%28%29) which is the orig name for this function. But, maybe it's something coming from a dependency.

**Comment by morinel** [comment]
Which version of gcmpush are you using?

**Comment by skypanther** [comment]
We're using a custom build based off of your 1.4 version that includes a fix for handling pushes when in the foreground. See https://github.com/skypanther/gcmpush with the key change in [GCMIntentService](https://github.com/skypanther/gcmpush/commit/b472fb5ba7a9dd426456de377b51d6878f35bb79#diff-f86bef3a5e2a0e2ab44ceee64b8e15f8R340)

I didn't realize you had released a 1.5 version. Looking at the diff, it doesn't appear that it would have any relation to the crash we're seeing. I will give it a try and bring our fork up to date.

**Comment by morinel** [comment]
Is still an issue?
